channels/channel.nos/nos2010/chn_nos2010.py

```diff
@@ -1331,6 +1332,81 @@ def update_video_item_live(self, item):
         item.complete = True
         return item
 
+    def create_iptv_streams(self, parameter_parser):
+        """ Fetch the available live channels using EPG endpoint and format them into JSON-STREAMS
+
+        :param ActionParser parameter_parser: a ActionParser object to is used to parse and
+                                                   create urls
+
+        :return: Formatted stations
+        :rtype: list
+        """
+
+        epg_data = JsonHelper.loads(UriHandler.open(datetime.datetime.now().strftime("https://start-api.npo.nl/epg/%Y-%m-%d?type=tv"),
+            no_cache=True,
+            additional_headers=self.__jsonApiKeyHeader))
+
+        parent_item = MediaItem("Live", "https://www.npostart.nl/live", media_type=mediatype.FOLDER)
+        items = []
+        iptv_streams = []
+
+        for stations in epg_data["epg"]:
+            livestream = stations["channel"]["liveStream"]
+            item = MediaItem(livestream["title"], livestream["shareUrl"], media_type=mediatype.VIDEO)
```

Comment:
You need to set `item.isLive=True` and perhaps `isGeoLocked` and `isDrmProtected` for these?

Reply:
isDrmProtected=True gives a kodi-warning (even with widevine installed). So I will leave that one out.